src/hearings/services/hearings.service.ts

```diff
@@ -73,15 +73,40 @@ export class HearingsService {
   }
 
   public submitHearingRequest(hearingRequestMainModel: HearingRequestMainModel): Observable<ResponseDetailsModel> {
-    return this.http.post<ResponseDetailsModel>('api/hearings/submitHearingRequest', hearingRequestMainModel);
+    let hearingRequestMainModel1 = hearingRequestMainModel;
+    const hearingRequestMainModel2: HearingRequestMainModel = {
+      ...hearingRequestMainModel,
+      hearingDetails: {
+        ...hearingRequestMainModel.hearingDetails,
+        hearingWindow: null
+      }
+    };
+
+    if (hearingRequestMainModel.hearingDetails.hearingWindow && Object.keys(hearingRequestMainModel.hearingDetails.hearingWindow).length === 0) {
+      hearingRequestMainModel1 = hearingRequestMainModel2;
+    }
+    return this.http.post<ResponseDetailsModel>('api/hearings/submitHearingRequest', hearingRequestMainModel1);
   }
 
   public updateHearingRequest(hearingRequestMainModel: HearingRequestMainModel): Observable<ResponseDetailsModel> {
+    let hearingRequestMainModel1 = hearingRequestMainModel;
```

Comment:
This code is duplicated. Sonar will complain. Can you please abstract into a function